nixos/modules/services/misc/nitter.nix

```diff
@@ -333,8 +350,12 @@ in
         after = [ "network-online.target" ];
         serviceConfig = {
           DynamicUser = true;
+          LoadCredential="guestAccountsFile:${cfg.guestAccounts}";
```

Comment:
Mostly to avoid storing the credentials in the nix store

For some reason, Nitter upstream doesn't have the credential path file as a part of the overall config, and only reads the path defined in the ENV variable.

For folks with a proper secret management setup via agenix or something similar, it should be easy to specify a file path.

For those who don't care about it being present in the nix store, they can just write a file and provide the path

This is also helpful to work around some chown shenanigans that `DynamicUser` creates (where you have a chicken and egg problem for trying to make the file visible to the systemd unit, since the user only exists when the service is started, but the service can't start without the file being present).

I borrowed this from https://github.com/NixOS/nixpkgs/blob/9ee63abe2cbeea5bf15f319a0a5aaf0919fe22e8/nixos/modules/services/networking/nix-serve.nix#L88

Open to any other sane changes as well, can't promise this is the definitive way to do this

Reply:
Ah right, `DynamicUser` does create problems with secret management. I guess `LoadCredential` is the most sane way to work around this. Looks good!